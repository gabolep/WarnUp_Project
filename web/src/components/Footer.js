import { Component } from 'react';
import React from 'react';

const styles = {
    footer: {
        backgroundColor: '#303030',
        color: '#fff',
        flexDirection: 'column',
        alignItems: 'center',
        height: '100px',
        position: 'absolute',
        bottom: '0',
        marginTop: '100px',
        textAlign: 'center',
        width: '100%'

    },
    author: {
        paddingTop: '20px',
        textAlign: 'center',
        fontWeight: '630',
        fontSize: '1rem',
    }
}
class Footer extends Component {
    render(){
        return(
            <footer data-testid="footer" style = {styles.footer} > 
                <h3 data-testid= "h3" style = {styles.author}>website created by Gabriel Escudero</h3>
            </footer> 
        )
    }
}



export default Footer;
