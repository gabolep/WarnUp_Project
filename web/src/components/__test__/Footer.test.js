import react from "react";
import ReactDOM from "react-dom";
import Footer from '../Footer';

import { render, cleanup } from '@testing-library/react'
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Footer></Footer>, div)
})
it("render footer correctly", () => {
    const { getByTestId } = render(<Footer/>)
    expect(getByTestId("h3")).toHaveTextContent("website created by Gabriel Escuder");

})
it("matches snapshot", () => {
    const tree = renderer.create(<Footer></Footer>).toJSON();
    expect(tree).toMatchSnapshot();
})