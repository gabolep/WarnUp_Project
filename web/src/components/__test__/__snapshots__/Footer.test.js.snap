// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<footer
  data-testid="footer"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#303030",
      "bottom": "0",
      "color": "#fff",
      "flexDirection": "column",
      "height": "100px",
      "marginTop": "100px",
      "position": "absolute",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <h3
    data-testid="h3"
    style={
      Object {
        "fontSize": "1rem",
        "fontWeight": "630",
        "paddingTop": "20px",
        "textAlign": "center",
      }
    }
  >
    website created by Gabriel Escudero
  </h3>
</footer>
`;
